pagefaultgames/pokerogue
# [Refactor]  Refactored Game Settings UI navigation menu and sorting

## What are the changes?
Renamed the "Accessibility" settings to "Display"
Added "Audio" settings
Moved several settings to a more appropriate settings tab

## Why am I doing these changes?
Most of the settings currently labeled "Accessibility" are not truly accessibility settings.

The current "General" settings menu is large and contains a variety of different types of settings.
Currently there are 20 "General" settings, and 9 "Accessibility settings.
In this PR there are 9 "General" settings, 17 "Display" settings, and 3 "Audio" settings.

## What did change?
`navigationMenu.ts` was modified to add "Display" and "Audio" settings.
`ui-inputs.ts` and `ui.ts` were modified to accommodate the new "Display" and "Audio" Settings.
The new settings ui handlers were added to the `buttonCycleOption` whitelist in `ui-inputs.ts`.

`settings-accessiblity-ui-handler.ts` became `settings-audio-ui-handler.ts`.
`settings-display-ui-handler.ts` was added.

In `settings.ts`

- `Enable_Retries` became a GENERAL setting
- `Language`, `UI_Theme`, `Window_Type`, `Damage_Numbers`, `Move_Animations`, `Show_Stats_on_Level_Up`, `Sprite_Set`, `Fusion_Palette_Swaps`, and `Player_Gender` became DISPLAY settings
- `Master_Volume`, `BGM_Volume`, and `SE_Volume` became AUDIO settings

### Screenshots/Videos
![Game-Settings](https://github.com/pagefaultgames/pokerogue/assets/167576411/9ae54733-1aaf-4285-a5fd-430ae61f5a41)
![Bottom-Of-General-Settings](https://github.com/pagefaultgames/pokerogue/assets/167576411/6a6f544c-087b-48ed-bb80-faedfd07d8ba)

![Display-Settings-1](https://github.com/pagefaultgames/pokerogue/assets/167576411/85d596df-81be-4510-9961-376e99a2a88a)
![Display-Settings-2](https://github.com/pagefaultgames/pokerogue/assets/167576411/e6f9a8ff-4bdb-411f-af88-c187212badff)
![Display-Settings-3](https://github.com/pagefaultgames/pokerogue/assets/167576411/7f7f2c0d-e617-4874-8b80-052964a1a44e)

![Audio-Settings](https://github.com/pagefaultgames/pokerogue/assets/167576411/145c0976-932b-4b17-9fc0-4cf5a73277f0)

## How to test the changes?
Open the game and verify that the settings are in the appropriate tab.
Verify that you are able to navigate the settings tabs as expected.
Verify that that all current settings still work.

## Checklist
